### DISCUSSION TITLE
questions about the case_study and practical usage

### DISCUSSION THREAD

**Discussion by sergey-semenov-ai** [RUCAIBox/RecBole#1105] [Q&A]
Hello!
I have 3 questions about the practical usage of RecBole.
Let's take the BPR model and the ml-100k dataset as an example.
1. Do I understand correctly, recbole.utils.case_study.hat full_sort_topk evaluates the preferences of selected users on the entire daset, and not only on test_data?
2. full_sort_topk returns predictions of movies that the user has not watched yet?
3. I trained the model. After the training, several new users added their ratings to the films. How can I get recommendations for new users using an already trained model?

**Comment by Wicknight** [comment]
Hello!
1. Just as you said, we evaluate on the entire dataset, not only on test_data.
2. Just as you said, we will set the score of historical movies to negative infinity so we will return movies that user has not watched yet.
3. This is a cold start problem. At present, there is no special fixed practice. You can refer to issue #952.
